File: examples/asyncapi/provider/src/main/java/io/example/asyncapi/provider/web/mappers/CustomerDTOsMapper.java
package io.example.asyncapi.provider.web.mappers;

import io.example.asyncapi.provider.domain.*;
import io.example.asyncapi.provider.service.dtos.*;
import io.example.asyncapi.provider.web.model.*;
import java.math.*;
import java.time.*;
import java.util.*;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;
import org.springframework.data.domain.Page;

@Mapper(uses = BaseMapper.class)
public interface CustomerDTOsMapper {

    CustomerDTOsMapper INSTANCE = Mappers.getMapper(CustomerDTOsMapper.class);

    // request mappings
    Customer asCustomer(CustomerDTO dto);

    // response mappings

    List<CustomerDTO> asCustomerDTOList(List<Customer> entityList);

    CustomerPaginatedDTO asCustomerPaginatedDTO(Page<Customer> page);

    default Page<CustomerDTO> asCustomerDTOPage(Page<Customer> page) {
        return page.map(this::asCustomerDTO);
    }

    CustomerDTO asCustomerDTO(Customer entity);
}

File: examples/kustomer-address-jpa/src/main/kotlin/io/zenwave360/examples/kotlin/web/mappers/CustomerDTOsMapper.kt
package io.zenwave360.examples.kotlin.web.mappers

import io.zenwave360.examples.kotlin.web.mappers.*
import io.zenwave360.examples.kotlin.domain.*
import io.zenwave360.examples.kotlin.service.dtos.*
import io.zenwave360.examples.kotlin.web.dtos.*

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.factory.Mappers
import java.math.*
import java.time.*
import java.util.*
import org.springframework.data.domain.Page

@Mapper(uses = [BaseMapper::class])
interface CustomerDTOsMapper {

    companion object {
        val INSTANCE: CustomerDTOsMapper = Mappers.getMapper(CustomerDTOsMapper::class.java)
    }

    // request mappings
    fun asCustomerSearchCriteria(dto: CustomerSearchCriteriaDTO): CustomerSearchCriteria
    fun asCustomer(dto: CustomerDTO): Customer

    // response mappings

    fun asCustomerDTOList(entityList: List<Customer>): List<CustomerDTO>
    @Mapping(target = "content", source = "content", conditionExpression = "java(page.getContent() != null)")
    fun asCustomerPaginatedDTO(page: Page<Customer>): CustomerPaginatedDTO
    fun asCustomerDTOPage(page: Page<Customer>): Page<CustomerDTO> {
        return page.map { this.asCustomerDTO(it) }
    }

    fun asCustomerDTO(entity: Customer): CustomerDTO


}
